### PR TITLE
Support composer 2

### LINKED ISSUE #218: upgrade composer api version
Composer just merge the branch 2.0 into master.

So we got this issue :
    - hirak/prestissimo 0.3.10 requires composer-plugin-api ^1.0.0 -> found composer-plugin-api[2.0.0] but it does not match your constraint.

### PR DESCRIPTION
**Pull request by eiriksm**
This fixes #218 

I just disabled the plugin when a certain class constant was removed (indicating that composer 2 is used).

### CODE REVIEW

File: src/Plugin.php [hunk 1]
```diff
@@ -63,6 +63,10 @@ public function activate(Composer $composer, IO\IOInterface $io)
         }
         // @codeCoverageIgnoreEnd
 
+        if (!defined('\Composer\Installer\InstallerEvents::POST_DEPENDENCIES_SOLVING')) {
```
Comment (kubawerlos) [inline src/Plugin.php:66]: What is the reason for this condition?

File: src/Plugin.php [hunk 2]
```diff
@@ -97,6 +101,9 @@ class_exists(__NAMESPACE__ . '\\' . $class);
 
     public static function getSubscribedEvents()
     {
+        if (!defined('\Composer\Installer\InstallerEvents::POST_DEPENDENCIES_SOLVING')) {
```
Comment (kubawerlos) [inline src/Plugin.php:104]: What is the reason for this condition?

File: src/Plugin.php (hunk 1 above)
Comment (eiriksm) [reply src/Plugin.php:66]: It indicates that composer 2 is used (as stated)? :)

Comment (kubawerlos) [reply src/Plugin.php:66]: But for this `PluginInterface::PLUGIN_API_VERSION` was introduced.

**Comment by heddn** [comment]
This is important now as I have CI builds failing now that https://github.com/composer/composer/releases/tag/2.0.0 is released.

**Comment by Levivb** [comment]
@heddn depending on your setup you could remove this plugin from your CI pipeline as composer 2 natively downloads necessary files in parallel

**Comment by heddn** [comment]
I can and will, but I have several legacy jobs that haven't updated yet and a simple fix here would let them continue to run without interruption.

**Comment by hirak** [comment]
I won't fix this plugin anymore, just use Composer 2.
https://github.com/composer/composer/pull/5293

**Pull request closed without merging**